### PR TITLE
Updating the documentation style guide

### PR DESCRIPTION
**Pull request by manez**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/1040

# What does this Pull Request do?

Moves the documentation style guide from the technical documentation section to the contributing section (I think it makes more sense there), and adds some guidelines, mostly based on the guidelines we documented for the [current sprint](https://docs.google.com/document/d/1wz3kkjUoZDhMfI8GDaql5o6H2jQanul48cBr8LL4aSI).

# What's new?
Deleted the old page, put it in a new spot, added some items. 

# How should this be tested?

Preview the document, see if it makes sense/is helpful as a guide for documentation contributors. 

# Interested parties
@Islandora/8-x-committers and @ysuarez

### CODE REVIEW

**Comment by whikloj** [comment]
@manez does this need a line in `mkdocs.yml` too?

**Comment by manez** [comment]
@whikloj Indeed it does. I moved it to the new section in `mkdocs.yml`

File: mkdocs.yml [hunk 1]
```diff
@@ -77,4 +76,5 @@ pages:
       - 'Islandora 7': 'technical-documentation/migrate-7x.md'
   - Contributing:
       - 'How to contribute': 'contributing/CONTRIBUTING.md'
+      - 'Documentation Style Guide': 'technical-documentation/docs_style_guide.md'
```
Comment (whikloj) [inline mkdocs.yml]: path needs to be `'contributing/docs_style_guide.md'`

Comment (manez) [reply mkdocs.yml]: 🤦‍♀

**Review by whikloj — APPROVED** [review]
👍

**Pull request merged**